azurerm/resource_arm_private_dns_zone_test.go

```diff
@@ -0,0 +1,219 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+)
+
+func TestAccAzureRMPrivateDnsZone_basic(t *testing.T) {
+	resourceName := "azurerm_private_dns_zone.test"
+	ri := tf.AccRandTimeInt()
+	config := testAccAzureRMPrivateDnsZone_basic(ri, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateDnsZoneDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateDnsZoneExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateDnsZone_requiresImport(t *testing.T) {
+	if !requireResourcesToBeImported {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_private_dns_zone.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateDnsZoneDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateDnsZone_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateDnsZoneExists(resourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMPrivateDnsZone_requiresImport(ri, location),
+				ExpectError: testRequiresImportError("azurerm_private_dns_zone"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateDnsZone_withTags(t *testing.T) {
```

Comment:
This test is currently failing for me locally.

The tags get correctly applied but don't seem to be in the state, not sure what is going on here.

Any help appreciated

Reply:
It's happening because the response from the API only has the old values, its returning stale data.

Reply:
I've added a sleep, unless anyone has any better ideas...